trie/interface.go

```diff
@@ -46,8 +46,8 @@ type node interface {
 	getValue() []byte
 
 	commitDirty(level byte, maxTrieLevelInMemory uint, originDb common.DBWriteCacher, targetDb common.DBWriteCacher) error
-	commitCheckpoint(originDb common.DBWriteCacher, targetDb common.DBWriteCacher, checkpointHashes CheckpointHashesHolder, leavesChan chan core.KeyValueHolder, ctx context.Context, stats common.SnapshotStatisticsHandler) error
-	commitSnapshot(originDb common.DBWriteCacher, leavesChan chan core.KeyValueHolder, ctx context.Context, stats common.SnapshotStatisticsHandler) error
+	commitCheckpoint(originDb common.DBWriteCacher, targetDb common.DBWriteCacher, checkpointHashes CheckpointHashesHolder, leavesChan chan core.KeyValueHolder, ctx context.Context, stats common.SnapshotStatisticsHandler, idleProvider IdleNodeProvider) error
```

Comment:
can you check if the idleNodeProvider can be added on the receiver, instead of as parameter, as this does not change.

Reply:
can be done but the resulting trie implementations will all require this idleNodeProvider:
For example, the decodeNode function will need this parameter, then, all 3 implementations of the trie nodes, the patriciaMerkleTrie, interceptedTrieNode (along with the interceptors - and maybe the trie nodes resolvers). 
This will add significant complexity when only the snapshot/checkpoint process needs the process control